Source/WebKit/NetworkProcess/cocoa/NetworkTaskCocoa.mm

```diff
@@ -251,6 +251,13 @@ static bool shouldCapCookieExpiryForThirdPartyIPAddress(const WebCore::IPAddress
 #endif
 }
 
+void NetworkTaskCocoa::updateTaskWithStoragePartitionIdentifier(const WebCore::ResourceRequest& request)
+{
+    // FIXME: Remove respondsToSelector when available with NWLoader. rdar://134913391
+    if (auto* networkStorageSession = m_networkSession->networkStorageSession(); networkStorageSession && networkStorageSession->isOptInCookiePartitioningEnabled() && [task() respondsToSelector:@selector(set_storagePartitionIdentifier:)])
```

Comment:
Nit: the condition is too complicated to be put in one-line; we could add early return for check failure, e.g. `if (!networkStorageSession) return`.
Also we should use `CheckedPtr`

Reply:
If `isOptInCookiePartitioningEnabled` is enabled, but cookie policy is allowing all cookies, do we need to set partition identifier?

Reply:
Yes, we should still respect the `Partitioned` attribute of the cookie and put it into the correct partition, even if we are accepting all cookies.

Reply:
My question is more about whether the request is allowed to use non-partitioned cookies (i.e. search in non-partitioned cookie storage) if the partitionIdentifier is set?

Reply:
Currently, the CFNetwork implementation will allow access to both partitioned and unpartitioned cookies. I have a follow-up patch that restricts access to only partitioned cookies. I want this patch to focus on setting the partitionIdentifier.